### PR TITLE
Extend clean_non_existing_packs

### PR DESCRIPTION
**Pull request by moishce-zz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-631)

## Description
Extend clean_non_existing_packs for more cases in the upload flow

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/Marketplace/upload_packs.py [hunk 1]
```diff
@@ -217,17 +217,17 @@ def clean_non_existing_packs(index_folder_path: str, private_packs: list, storag
     Returns:
         bool: whether cleanup was skipped or not.
     """
-    if ('CI' not in os.environ) or (
-            os.environ.get('CI_COMMIT_BRANCH') != 'master' and storage_bucket.name == GCPConfig.PRODUCTION_BUCKET) or (
-            os.environ.get('CI_COMMIT_BRANCH') == 'master' and storage_bucket.name not in
-            (GCPConfig.PRODUCTION_BUCKET, GCPConfig.CI_BUILD_BUCKET)):
-        logging.info("Skipping cleanup of packs in gcs.")  # skipping execution of cleanup in gcs bucket
-        return True
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/upload_packs.py:225]: Can you elaborate on what is this part you removed here?

**Review by ShahafBenYakir — COMMENTED** [review]
Look OK to me.
@Noy-Maimon Please verify me

File: Tests/Marketplace/upload_packs.py (hunk 1 above)
Comment (moishce) [reply Tests/Marketplace/upload_packs.py:225]: I removed the code that makes it not run in tests (it was part of the issue)

**Review by ShahafBenYakir — APPROVED** [review]

**Review by Noy-Maimon — COMMENTED** [review]
Looks good 👏 
How was this tested?
A suitable test would be:
1. creating a new dummy pack in this branch
2. running the upload flow on that branch
3. once that upload is done, take that target bucket link (`should look like upload-flow/builds/<branch_name>/<build_number>/content`)
4. delete that dummy pack from the branch
5. run the upload again, but with the argument storage_base_path as the path from clause 3
6. verify that the dummy pack is missing from the index and the bucket

**Pull request closed without merging**